compiler/src/dmd/backend/x86/cod3.d

```diff
@@ -30,6 +30,7 @@ import dmd.backend.cc;
 import dmd.backend.cdef;
 import dmd.backend.cgcse;
 import dmd.backend.code;
+import dmd.backend.arm.cod3;
 import dmd.backend.x86.code_x86;
```

Comment:
I can't see you getting far mixing different architectures like this.

Reply:
In case this comes across as an inactionable comment, to clarify.  Changes like this make it quite clear that a lot more work needs to go into separation of IR and target modules in the back-end before the introduction of any new architecture can be done (sans this kind of spaghetti imports).

That alone would make introducing any new foreign architecture into dmd more palatable.

Reply:
A good chunk of the code between the two will be identical. It's a balance between having them being totally separate and sharing common code. I don't know just where that balance will be, it will become apparent after I make more progress.

Reply:
Having every architecture depend on every other architecture is not sustainable. Isn't contributing a new architecture hard enough without technical debt like this?

Reply:
It won't be clear where the separation should be until I make much more progress.